homeassistant/helpers/config_validation.py

```diff
@@ -297,9 +297,12 @@ def icon(value: Any) -> str:
             "minutes": vol.Coerce(int),
             "seconds": vol.Coerce(int),
             "milliseconds": vol.Coerce(int),
+            "microseconds": vol.Coerce(int),
         }
     ),
-    has_at_least_one_key("days", "hours", "minutes", "seconds", "milliseconds"),
+    has_at_least_one_key(
+        "days", "hours", "minutes", "seconds", "milliseconds", "microseconds"
```

Comment:
Why are we adding microseconds ? And how is that related to adding collections to template? Please split out this change for a new PR.

Reply:
The idea was to allow the `timedelta` to be serialized without doing unit conversion. 

`timedelta` natively stores its data as days, seconds, and microseconds. By adding the above, I don't need to do anything like "microseconds / 1000" which could lead to rounding issues.